src/components/form/preview.vue

```diff
@@ -84,6 +86,53 @@ fetchForm();
 const handleSubmit = () => {
   previewModal.show();
 };
+
+
+
+/**
+ * Web Form expects host application to provide a function that it can use to
+ * fetch attachments. Signature of the function is (url) => Response; where
+ * Response is subset of web standard  {@link Response}.
+ */
+const getAttachment = (url) => request({
+  url: apiPaths.formAttachment(
+    props.projectId,
+    props.xmlFormId,
+    props.draft,
+    url.pathname.substring(1)
+  ),
+  alert: false
+}).then(axiosResponse => {
+  const { data, status, statusText, headers } = axiosResponse;
+
+  const fetchHeaders = new Headers();
+  for (const [key, value] of Object.entries(headers)) {
+    if (key === 'content-type') {
+      // because web-forms doens't want space between media type and charset
+      // https://github.com/getodk/web-forms/pull/259#discussion_r1887227207
+      fetchHeaders.append(key, value.replace('; charset', ';charset'));
+    } else {
+      fetchHeaders.append(key, value);
+    }
+  }
+
+  let body;
+  if (typeof (data) === 'string') {
+    body = data;
+  } else if (headers['content-type'].includes('application/json') ||
+             headers['content-type'].includes('application/geo+json')) {
+    body = JSON.stringify(data);
+  } else {
+    // eslint-disable-next-line no-console
+    console.error('response data is not a known text format');
+  }
+
+  return new Response(body, {
+    status,
+    statusText,
+    headers: fetchHeaders,
+  });
```

Comment:
I looked for any option to get [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response) object directly from Axios but couldn't find one. They have recently [implemented](https://github.com/axios/axios/releases/tag/v1.7.0-beta.0) `fetch` adaptor but doesn't provide any way to get original fetch Response